File: examples/python/sqs_send_message.py
"""Envía un mensaje a una cola SQS.

Uso: python3 sqs_send_message.py <queue-url> <texto-del-mensaje>
"""
import sys

import boto3

sqs = boto3.client(
    "sqs",
    endpoint_url="http://localhost:4566",
    aws_access_key_id="test",
    aws_secret_access_key="test",
    region_name="us-east-1",
)


def main():
    if len(sys.argv) < 3:
        raise SystemExit("Uso: python3 sqs_send_message.py <queue-url> <texto-del-mensaje>")
    queue_url = sys.argv[1]
    message_body = " ".join(sys.argv[2:])

    result = sqs.send_message(QueueUrl=queue_url, MessageBody=message_body)
    print("Mensaje enviado. MessageId:", result["MessageId"])


if __name__ == "__main__":
    main()

File: examples/rust/sqs_send_message.rs
// Envia un mensaje a una cola SQS.
// Uso: cargo run --bin sqs_send_message -- <queue-url> <texto-del-mensaje>

use aws_config::BehaviorVersion;
use aws_credential_types::Credentials;
use aws_sdk_sqs::{config::Region, Client, Config};

#[tokio::main]
async fn main() -> Result<(), aws_sdk_sqs::Error> {
    let args: Vec<String> = std::env::args().collect();
    if args.len() < 3 {
        eprintln!("Uso: cargo run --bin sqs_send_message -- <queue-url> <texto-del-mensaje>");
        std::process::exit(1);
    }
    let queue_url = &args[1];
    let body = args[2..].join(" ");

    let config = Config::builder()
        .behavior_version(BehaviorVersion::latest())
        .endpoint_url("http://localhost:4566")
        .region(Region::new("us-east-1"))
        .credentials_provider(Credentials::new("test", "test", None, None, "floci"))
        .build();
    let client = Client::from_conf(config);

    let result = client.send_message().queue_url(queue_url).message_body(&body).send().await?;
    println!("Mensaje enviado. MessageId: {}", result.message_id().unwrap_or("?"));
    Ok(())
}
